src/AWS.Deploy.CLI/Commands/GenerateDeploymentProjectCommand.cs

```diff
@@ -0,0 +1,185 @@
+            _awsResourceQueryer = awsResourceQueryer;
+            _deploymentBundleHandler = deploymentBundleHandler;
+            _commandLineWrapper = commandLineWrapper;
+            _session = session;
+        }
+
+        public async Task ExecuteAsync(string saveDirectory)
+        {
+            var orchestrator = GetOrchestrator();
+            var recommendations = await GenerateDeploymentRecommendations(orchestrator);
+            var selectedRecommendation = _consoleUtilities.AskToChooseRecommendation(recommendations);
+
+            if (string.IsNullOrEmpty(saveDirectory))
+                saveDirectory = GenerateDefaultSaveDirectory(_session.ProjectDefinition);
+
+            var directoryCreatedByDeployTool = ValidateSaveDirectory(saveDirectory);
+
+            var result = await CheckDirectoryUnderSourceControl(saveDirectory);
+            if (!result.Success)
+            {
+                _toolInteractiveService.WriteLine();
+                var YesNoResult = _consoleUtilities.AskYesNoQuestion("The target directory for saving the CDK " +
+                    "deployment project is not being tracked by a Git repository. Do you still want to continue?", YesNo.Yes);
+
+                if (YesNoResult == YesNo.No)
+                {
+                    if (directoryCreatedByDeployTool)
+                        Directory.Delete(saveDirectory);
+                    return;
+                }
+            }
+
+            await _cdkProjectHandler.CreateCdkProjectForDeployment(selectedRecommendation, _session, saveDirectory);
+
+            _toolInteractiveService.WriteLine();
+            _toolInteractiveService.WriteLine($"The CDK deployment project is saved at: {saveDirectory}");
+        }
+
+        private Orchestrator GetOrchestrator()
+        {
+            return new Orchestrator(
+                        _session,
+                        _orchestratorInteractiveService,
+                        _cdkProjectHandler,
+                        _cdkManager,
+                        _awsResourceQueryer,
+                        _deploymentBundleHandler,
+                        new DockerEngine.DockerEngine(_session.ProjectDefinition),
+                        new[] { RecipeLocator.FindRecipeDefinitionsPath() });
+        }
+
+        private async Task<List<Recommendation>> GenerateDeploymentRecommendations(Orchestrator orchestrator)
+        {
+            var recommendations = await orchestrator.GenerateDeploymentRecommendations();
+            if (recommendations.Count == 0)
+            {
+                throw new FailedToGenerateAnyRecommendations("The project you are trying to deploy is currently not supported.");
+            }
+            return recommendations;
+        }
+
+        private string GenerateDefaultSaveDirectory(ProjectDefinition projectDefinition)
+        {
+            var projectDirectoryFullPath = new DirectoryInfo(projectDefinition.ProjectPath).Parent.FullName;
```

Comment:
`DirectoryInfo` constrictor can throw exception. We should guard against it.

Reply:
Added a wrapper method called `GetDirectoryInfo` that handles any exceptions thrown by the `DirectoryInfo` constructor.